### PR TITLE
DOCSP-32086 Convert Stored Procedures Content

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

Task page, how to convert a stored procedure to MongoDB syntax with the query converter.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-32086/code-generation/query-converter/convert-stored-procedures/

## JIRA

https://jira.mongodb.org/browse/DOCSP-32086

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=654b9bc45b686d6bd95b4fc8

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 1]
```diff
@@ -10,3 +10,276 @@ Convert Stored Procedures
...
+         
+         The SQL syntax displays in the 
+         :guilabel:`Imported Stored Procedure` pane.
+
+      #. Select a :guilabel:`Target Language` from the drop-down.
+         
+         Query converter supports stored procedure conversions to
+         ``Javascript``, ``C#``, and ``Java``.
+
+      #. Click the :guilabel:`Convert` button. Wait for the query
```
Comment (ianf-mongodb) [inline source/code-generation/query-converter/convert-stored-procedures.txt:89]: Not sure if we want to go into details about what is in and out of scope for conversions here or just remain procedural. I did get the following error when attempting to convert [a SP which referenced a temporary table ](https://gist.github.com/ianf-mongodb/da452c525389d142bec137abae78bb51):

Screenshot 2023-11-07 at 11 28 32 AM

File: source/code-generation/query-converter/convert-queries.txt [hunk 2]
```diff
@@ -35,11 +35,11 @@ Steps
 
    .. step:: Open the SQL query converter view
 
-      If it is your first time using the query converter in your 
-      project, click :guilabel:`Paste SQL Query`.
-      If you have previously converted SQL code in your project, click the
-      :guilabel:`+ ADD` button on the left-hand pane next to 
-      :guilabel:`Queries`.
+      - If it is your first time using the query converter in your
```
Comment (ianf-mongodb) [inline source/code-generation/query-converter/convert-queries.txt:38]: **IGNORE**

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 3]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+
+      a. Specify the connection details to your source database. 
+         
+         For details, see :ref:`rm-relational-database-connection-strings`.
+      #. Click :guilabel:`Connect`.
+
+   .. step:: Select stored procedures
+
+      a. On the :guilabel:`Import Database Objects` modal, click
+         the :icon-fa5:`chevron-right` icon next :guilabel:`Database`.
```
Comment (mdb-ashley) [inline source/code-generation/query-converter/convert-stored-procedures.txt:57]:
```suggestion
         the :icon-fa5:`chevron-right` icon next to :guilabel:`Database`.
```

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 4]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+      a. Click a stored procedure's name from the left-hand pane under
+         :guilabel:`Stored Procedures`.
+         
+         The SQL syntax displays in the 
+         :guilabel:`Imported Stored Procedure` pane.
+
+      #. Select a :guilabel:`Target Language` from the drop-down.
+         
+         Query converter supports stored procedure conversions to
+         ``Javascript``, ``C#``, and ``Java``.
```
Comment (mdb-ashley) [inline source/code-generation/query-converter/convert-stored-procedures.txt:88]: [question] Do we usually format programming language names in monospace? Question applies globally.

File: source/code-generation/query-converter/convert-stored-procedures.txt (hunk 1 above)
Comment (mdb-ashley) [reply source/code-generation/query-converter/convert-stored-procedures.txt:90]: I think this information would go in **About this Task** if we decide to include it.

**Review by mdb-ashley — APPROVED** [review]
LGTM % a couple of comments. Nice work, Ian!

File: source/code-generation/query-converter/convert-stored-procedures.txt (hunk 4 above)
Comment (ianf-mongodb) [reply source/code-generation/query-converter/convert-stored-procedures.txt]: After taking a look at some other repos, and the style guide it looks like the answer is no. I have updated accordingly. Thanks for bringing this up @mdb-ashley!

**Comment by ianf-mongodb** [comment]
This is ready for review @tomhollander, please see my comment WRT conversion error content.

File: source/code-generation/query-converter/convert-stored-procedures.txt (hunk 1 above)
Comment (tomhollander) [reply source/code-generation/query-converter/convert-stored-procedures.txt:90]: @ianf-mongodb Some of these error cases will be temporary as we are still making improvements. Probably no need to focus on the exact cases that do/don't work in the docs as they will change.

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 5]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
 
+You can import and convert your SQL stored procedures to MongoDB syntax 
+with the query converter. The query converter considers the mapping 
+rules and schema transformations defined in your project when converting
+your SQL code.
+
+About this Task
+---------------
+
+.. include:: /includes/fact-query-converter-generic.rst
```
Comment (tomhollander) [inline source/code-generation/query-converter/convert-stored-procedures.txt:21]: Sorry I know this was in the previous PR but I think we should review this text:
> Consider SQL complexity when using the query converter. The query converter cannot convert highly complex SQL syntax. If your SQL code isn't converting accurately, refactor it and try again.

The capabilities of the feature will change over time, and we aspire to be able to convert any code regardless of its complexity. If something doesn't work, "refactoring" is unlikely to be practical. Can we make this a bit more vague, along the lines of:
> The query converter uses AI technology and is not always perfect, especially for long or complex queries or stored procedures. Some queries may not be converted correctly, or they may not be converted at all.

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 6]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+
+         .. tip::
+
+            You can toggle stored procedures for conversion by clicking
+            the :icon-fa5:`check-square` icon next to a stored procedure's
+            name. All stored procedures are selected by default.
+
+      #. Click :guilabel:`Save`.
+
+         The SQL syntax for each stored procedure in your database
```
Comment (tomhollander) [inline source/code-generation/query-converter/convert-stored-procedures.txt:72]: Replace "SQL syntax" with "code"

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 7]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+            name. All stored procedures are selected by default.
+
+      #. Click :guilabel:`Save`.
+
+         The SQL syntax for each stored procedure in your database 
+         schema is imported into your project and is visible in the 
+         :guilabel:`Query Converter` pane under 
+         :guilabel:`Stored Procedures`.
+
+   .. step:: Convert and test syntax
```
Comment (tomhollander) [inline source/code-generation/query-converter/convert-stored-procedures.txt:77]: Again I think "Syntax" is the wrong word here.

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 8]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+
+      a. Click a stored procedure's name from the left-hand pane under
+         :guilabel:`Stored Procedures`.
+         
+         The SQL syntax displays in the 
+         :guilabel:`Imported Stored Procedure` pane.
+
+      #. Select a :guilabel:`Target Language` from the drop-down.
+         
+         Query converter supports stored procedure conversions to
```
Comment (tomhollander) [inline source/code-generation/query-converter/convert-stored-procedures.txt:87]: This warrants a longer explanation, probably outside of the procedure. MongoDB doesn't have stored procedures per se, so users need to figure out where they want to host this logic once converted. The main options are to host in their app's backend, or they could use App Services Functions. Users should choose the programming language that matches the environment where the code will be hosted. The converted code will use the selected language and have MongoDB queries/aggregations interleaved.

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 9]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+         Javascript, C#, and Java.
+
+      #. Click the :guilabel:`Convert` button. Wait for the query 
+         converter to convert your code. 
+
+         The converted MongoDB syntax displays
```
Comment (tomhollander) [inline source/code-generation/query-converter/convert-stored-procedures.txt:93]: It's code, not syntax.

File: source/code-generation/query-converter/convert-stored-procedures.txt [hunk 10]
```diff
@@ -10,3 +10,277 @@ Convert Stored Procedures
...
+         :guilabel:`Converted MongoDB Query` pane to copy the MongoDB 
+         syntax to your clipboard.
+
+      #. Test and validate the generated MongoDB syntax.
+
+         .. tip::
+
+            .. include:: /includes/fact-query-converter-filter.rst
+
+Examples
```
Comment (tomhollander) [inline source/code-generation/query-converter/convert-stored-procedures.txt:106]: Nice examples.

**Review by tomhollander — CHANGES_REQUESTED** [review]
Nice start!

File: source/code-generation/query-converter/convert-stored-procedures.txt (hunk 8 above)
Comment (ianf-mongodb) [reply source/code-generation/query-converter/convert-stored-procedures.txt:96]: [I added a version of your blurb to the ](https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-32086/code-generation/query-converter/convert-stored-procedures/#before-you-begin) `Before you Begin` section outlining these major points and pointing the the app function docs, LMKYWT.

**Review by ianf-mongodb — COMMENTED** [review]
This is ready for another look @tomhollander, 👍 = updated.

**Review by tomhollander — APPROVED** [review]
Looks good, thanks!
A quick heads-up - we realised that Oracle treats "functions" and "packages" as distinct object types to stored procedures. We don't handle these at all now, but we plan on adding support soon. The procedure will be identical for stored procs, but we will group these objects under their own heading. This topic is fine now but we should think about how to have it include functions and packages in the future.

**Pull request merged**